resources/lib/navigation/player.py

```diff
@@ -175,9 +179,10 @@ def get_upnext_info(videoid, current_episode, metadata):
     else:
         next_info['play_info'] = {'play_path': common.build_url(videoid=next_episode_id,
                                                                 mode=g.MODE_PLAY)}
-    if 'creditsOffset' in metadata[0]:
-        next_info['notification_time'] = (metadata[0]['runtime'] -
-                                          metadata[0]['creditsOffset'])
+    try:
+        next_info['notification_offset'] = metadata[0]['creditsOffset']
+    except KeyError:
+        return None
```

Comment:
Not sure if this is what you want here?
It would mean that in case `creditsOffset` is missing `next_info` would be empty.

I would replace this code block like this:
```python
    if metadata[0].get('creditsOffset'):
        next_info['notification_offset'] = metadata[0]['creditsOffset']
```

The reason why I used a try-block in my attempt was because I needed to catch the RunTimeError exception for `xbmc.Player().getTotalTime()` anyway and in that case catching the `KeyError` was cleaner.

Reply:
I wanted to avoid the "if" because i think more than 90% of case the key exists
so i can change this with "pass"
<pre>
try:
        next_info['notification_offset'] = metadata[0]['creditsOffset']
    except KeyError:
        Pass
</pre>

Reply:
Why do you want to avoid the `if`? Exception handling also incurs overhead which is possibly greater than checking a key in a dictionary.

Reply:
But don't get me wrong, I don't think it really matters, either way.

Reply:
leave as original